internal/pkg/agent/application/monitoring/v1_monitor.go

```diff
@@ -897,6 +772,57 @@ func (b *BeatsMonitor) getServiceComponentProcessMetricInputs(
 	return inputs
 }
 
+// processorsForAgentFilestream returns processors used for agent logs in a filestream input.
+func processorsForAgentFilestream() []any {
+	processors := []any{
+		// drop all events from monitoring components (do it early)
+		// without dropping these events the filestream gets stuck in an infinite loop
+		// if filestream hits an issue publishing the events it logs an error which then filestream monitor
+		// will read from the logs and try to also publish that new log message (thus the infinite loop).
+		dropEventsFromMonitoringComponentsProcessor(),
+		// drop periodic metrics logs (those are useful mostly in diagnostic dumps where we collect log files)
+		dropPeriodicMetricsLogsProcessor(),
+	}
+	// if the event is from a component, use the component's dataset
```

Comment:
I see an `if` in the comment but the following appends happen always, am I missing something?

Reply:
The append always happens, but some of the added processors are conditional. The comment describes what all the added processors do collectively. Do you think it's unclear?